sawmill-core/src/main/java/io/logz/sawmill/SawmillConfiguration.java

```diff
@@ -0,0 +1,5 @@
+package io.logz.sawmill;
+
```

Comment:
@templatevk the discussion is important, it's good we are having it.

I disagree about this being unnecessary. When someone in the community looks through the history of these commits to find out how configuration objects are used, I want it to be clear that this is a generic feature, and not something added specifically for GeoIP.

That is why it should be in its own PR, with tests verifying that it works as expected.
If we were to choose to remove GeoIP from the codebase for some reason,  SawmillConfiguration will not have any tests, right?

This is a good indication that introducing the config object should be its own PR - it is a new feature of the library.

Reply:
Hello @barakm. Let me point out my view on what is happening here. 

You are talking about the history of commits and I agree with you, that commits should be granular and contain small changes that will show how the code evolved over the time.  Here we are talking about PR, so my suggestion that you see the PR as 1 to 1 to commit. So 1 PR can have only one commit. 

Here I have alternative vision. I think it is more appropriate to have granular commits in one PR and still one PR for holistic change like delivering one piece of functionality.

Also you didn't mention any criteria of PR's size. If it is so important, we have to introduce new development practice of, let's say, make PR not more than 300 lines of changes. Better something smarter than this.

I hope you understand that requirement to split PR from one person is considered as subjective and can be treated as preconception. So let's have a wider discussion and agree to spread it to the whole company.

P.S. I am aware of PR Guidelines document but there are no good criteria of PR size.

Reply:
I would also agree to have one PR with 3 granular commits instead of having 3 PRs that I'm waiting like a week for people to find time to review constantly pinging them.